### ISSUE TITLE
BadgeDotColors aren't deterministic

### ISSUE THREAD

**Issue by ghost** [hossein-zare/react-native-dropdown-picker#650]
Seems like they change quite randomly actually, for example if you pass three possible items, and select all three - two of them may be the same badge dot color despite passing through three badgeDotColors.

Was this the intended functionality? Ideally we should be able to define the individual badge dot color for each item in items

**Comment by hossein-zare** [comment]
>Was this the intended functionality?

Not randomly, Based on ASCII Code.
> Ideally we should be able to define the individual badge dot color for each item in items

You can add the feature and submit a PR.

**Comment by yaylinda** [comment]
Yeah, I am also trying to use the `badgeDotColors` and find it the ASCII code mapping to be not very intuitive. 

For my use-case, each item's value is a UUID string. Adding together the ascii code values for each character in the `item.value` string is kind of meaningless, and could cause collisions. 

Maybe the `badgeDotColors` prop could be a `Map<string, string>`, mapping item value to a hex code. Or the prop could simply be a callback function that the user could define and pass to it.

**Comment by bahaaTuffaha** [comment]
I tried to modify the code to just view colors same order as the array but its not working as it should be 🥲... have you guys found any solution to this?

**Comment by warmfire540** [comment]
i'm going to implement this in my fork - basically the two bade props will use order based indexing to match up w/ what you're putting in, can use if you'd like or patch your own
